### DISCUSSION TITLE
How to normalize re-called peaks for each cell type

### DISCUSSION THREAD

**Discussion by danielcgingerich** [stuart-lab/signac#469] [Q&A] [answered]
I have 23 snATAC datasets from 23 different samples (12 control, 11 stim).  Each dataset contains 6 cell types - excitatory/inhibitory neurons, astrocyte, microglia, oligodendrocyte, opc's.  

Initially, I RunTFIDF on each dataset individually before finding differentially accessible peaks.  

for each of the 23 datasets, I split the objects by cell type.  Then, pooled each cell type together (6 pools total) and fed the fragments to MACS2 peak calling.  From this, I created 6 new GRanges objects and quantified fragments for each cell type in each dataset (6 x 23 = 138 total matrices).  

Now, my question is, how do I normalize (TFIDF) these objects?  Initially, the datasets were normalized as a whole - so all 6 celltypes were taken into account for global scaling.  Once the data is separated by cell type and fragments are re-counted on a different set of peaks, the data is no longer normalized.  Initially, the whole dataset- all 6 cell types- were normalized together.  Now that I have only a fraction of the dataset in each matrix, I worry that the normalization will not be adequate because it doesnt take into account the other cells present in the data for global scaling.  Is it ok to just runTFIDF on these objects? I wonder if I should quantify fragments of all cells in each dataset for each cell type specific peak set, normalize as a whole, and then subset the cell types I need.

**Comment by timoast** [comment, marked as answer]
Rather than splitting the data by cell type and performing peak calling separately, I'd suggest running `CallPeaks()` on the original (non-split) object and setting the `group.by` parameter in `CallPeaks()`. This will automatically call peaks separately for each cell type, but will also combine the resulting peak sets so that you have a common set of peaks across all cell types in the dataset.

After peak calling and quantification, you could then split into the different cell types if you prefer, although it's not necessary and I would keep things in a single object. You can then find differentially accessible peaks between different groups (either between cell types, or between control and stimulated within a cell type), and include batch as a latent variable in the model.

**Reply by danielcgingerich** [reply]
@timoast This is what I will do going forward thank you. 

Currently, I am facing a deadline (tomorrow lol) and have re-called peaks on object subsets.  is it ok to RunTF-IDF on the cell types independently?  The only difference I see would be inverse document frequency. For differential accessibility, I dont see this having much of an affect

**Reply by timoast** [reply]
Right, the inverse document frequency is the only difference. If you are comparing within a cell type (ie, subset the cells, perform TF-IDF on the subset, then do DA on the subset), there is no difference since all the cells still use the same IDF value within the subset (it would only change results if running TF-IDF on different subsets and comparing them). I'm guessing here you're only comparing within a subset, since peak calling was done independently on each subset, so I'd say it does not matter if you do TF-IDF on the whole or subset in your case.
